src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
> the memory is cleared and starts fresh

Does this mean that connections made to Riak during one HTTP request / response cycle get torn down, every time?